ontology/yaml/resources/PLUMBING/entity_types/WSR.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### CANONICAL TYPES
+
+WSR_STD:
+  description: "Standard water softener."
+  is_canonical: true
+  implements:
+  - WSR
+  uses:
+  - run_status
+  opt_uses:
+  - lost_power_alarm
+  - water_flowrate_status
+  - run_mode
+
+WSR_AL:

```

Comment:
same comments as above, plus remove AL. instead, consolidate these into one type and make the alarms optional, then you can reuse this same type for all of these devices.

Reply:
noted on the alarms being optional to be more flexible.